### PR TITLE
feat: New field in TokenCriteria proto

### LINKED ISSUE #11588: Permissions values are not being applied as specified by user
# Bug Report

## Description

When a user specifies an amount in a permission is is being applied with a figure including several more decimal places. For example, 0.1 is being set as per the image below. This results in a larger amount than intended being required to meet the permission criteria.

image

## Steps to reproduce

Create a permission requiring 0.1 Eth

#### Expected behavior

Permission to be set with value entered by user.

#### Actual behavior

An amount larger than the value input by the user is being set.

### Additional Information

- Status desktop version: Master
- Operating System: Mac

### PR DESCRIPTION
**Pull request by endulab**
New field AmountInWeis is added to TokenCriteria proto.

AmountInWeis will have a wei-like units.
Amount field becomes deprecated because it kept string with float value.

Comparison (in case of Decimals == 5):
Amount (deprecated) = "1.2"
AmountInWeis = "120000"

Issue https://github.com/status-im/status-desktop/issues/11588

### CODE REVIEW

**Review by cammellos — APPROVED** [review]

**Review by mprakhov — APPROVED** [review]
LGTM

File: protocol/communities/community.go [hunk 1]
```diff
@@ -1477,6 +1479,37 @@ func hydrateChannelsMembers(communityID string, description *protobuf.CommunityD
...
+	for _, permission := range description.TokenPermissions {
+		for _, criteria := range permission.TokenCriteria {
+			if criteria.AmountInWeis != "" {
+				continue
+			}
+			// set AmountInWeis if missing
+			// Amount format (deprecated): "0.123"
+			// AmountInWeis format: "123000..000"
+			if criteria.Type == protobuf.CommunityTokenType_ERC20 {
+				criteria.AmountInWeis = floatToWeiIntFunc(criteria.Amount, criteria.Decimals)
```
Comment (jrainville) [inline protocol/communities/community.go]: Just a change to make it grammatically correct. It sounds better too haha
```suggestion
				criteria.AmountInWei = floatToWeiIntFunc(criteria.Amount, criteria.Decimals)
```

File: protocol/communities/manager.go [hunk 2]
```diff
@@ -4497,8 +4499,8 @@ func (m *Manager) AddCommunityToken(token *community_token.CommunityToken, clock
 			Type:              protobuf.CommunityTokenType_ERC721,
 			Symbol:            token.Symbol,
 			Name:              token.Name,
-			Amount:            "1",
```
Comment (jrainville) [inline protocol/communities/manager.go:4610]: Should we still add the `Amount` for old apps to still show the right value?

**Review by jrainville — COMMENTED** [review]
Nice job

File: protocol/communities/manager.go (hunk 2 above)
Comment (endulab) [reply protocol/communities/manager.go:4610]: Very good catch! We should.

**Review by jrainville — APPROVED** [review]
Looks good. I'd still appreciate if it was renamed `AmountInWei` :smile:

**Pull request merged**